### PR TITLE
Queue concurrency

### PR DESCRIPTION
**Pull request by justincy**
Fix for #747

### CODE REVIEW

**Comment by justincy** [comment]
Worked in node v0.10 but failed in v0.12 and iojs. I'll investigate.

File: lib/async.js [hunk 1]
```diff
@@ -834,8 +834,21 @@
                 if (q.tasks.length === q.concurrency) {
                     q.saturated();
                 }
-                async.setImmediate(q.process);
             });
+            setImmediate(q.process);
```
Comment (aearly) [inline lib/async.js]: Use async.setImmediate here, so it falls back to other methods when it's not available in the environment.

File: lib/async.js [hunk 2]
```diff
@@ -859,32 +872,22 @@
...
+                        var data = _map(tasks, function (task) {
+                            return task.data;
+                        });
+    
+                        if (q.tasks.length === 0) {
+                            q.empty();
+                        }
+                        workers += 1;
+                        var cb = only_once(_next(q, tasks));
+                        setImmediate(worker(data, cb));
```
Comment (aearly) [inline lib/async.js]: Likewise, `async.setImmediate`.

**Comment by justincy** [comment]
@aearly Fixed.

**Pull request merged**